zarr/storage.py

```diff
@@ -1779,7 +1779,13 @@ def __setitem__(self, key, value):
             self.zf.writestr(keyinfo, value)
 
     def __delitem__(self, key):
-        raise NotImplementedError
+        # raise NotImplementedError
+        try:
+            value = self[key]
+        except KeyError:
+            raise KeyError("Can not delete a nonexistent key")
+        
+        self[key] = b""
```

Comment:
Think we need to [hold a mutex]( https://github.com/zarr-developers/zarr-python/blob/2cfee9c0f0b2e0782f942ff0dd8681ecfb756ca0/zarr/storage.py#L1766 ) when writing. Otherwise yes, this seems conceptually correct.

Reply:
This is because I believe the set item method already handles the writing method. Is it really necessary to repeat the mutex?

Reply:
That's fair. Though this means we are acquiring the mutex twice. Once to check for the `key` in the `if` statement above and once to set the value. Maybe we can copy the code from those methods here and tidy it up so we acquire the mutex only once. What do you think?

Reply:
This change has been made